android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -452,9 +362,154 @@ public void onSignalingChange(PeerConnection.SignalingState signalingState) {
+    @Override
+    public void onTrack(final RtpTransceiver transceiver) {
+        Log.d(TAG, "onTrack");
+
+        RtpReceiver receiver = transceiver.getReceiver();
+        MediaStreamTrack track = receiver.track();
+
+        // Get The timestamp so that we can reorder the transceivers on the JS layer
+        Long tsLong = System.currentTimeMillis()/1000;
+        String ts = tsLong.toString();
+
+        if (track == null) {
+            return;
+        }
+        if(track.kind().equals(MediaStreamTrack.VIDEO_TRACK_KIND)){
+            videoTrackAdapters.addAdapter((VideoTrack) track);
+        }
+        remoteTracks.put(track.id(), track);
+
+        WritableMap params = Arguments.createMap();
+        WritableMap eventInfo = Arguments.createMap();
+        WritableArray streams = Arguments.createArray();
+        
+        eventInfo.putArray("streams", streams);
+        eventInfo.putMap("receiver", serializeReceiver(receiver));
+        eventInfo.putMap("transceiver", serializeTransceiver(transceiver));
+        eventInfo.putString("timestamp", ts);
+
+        // Getting the transceiver object associated with the receiver for the event
+        params.putInt("id", this.id);
+        params.putMap("info", eventInfo);
+
+        webRTCModule.sendEvent("peerConnectionOnTrack", params);
+    }
+
+    /*
+     * Triggered when a previously added remote track is removed by the remote
+     * peer, as a result of setRemoteDescription.
+     */
+    @Override
+    public void onRemoveTrack(RtpReceiver receiver){
+        // According to the W3C spec, we need to send out 
+        // the track Id so that we can remove it from the MediaStream objects stored
+        // at the JS layer, which are the same stream objects passed down to 
+        // the `track` event.
+
+        WritableMap params = Arguments.createMap();
+        params.putInt("id", this.id);
+        params.putString("trackId", receiver.track().id());
+        webRTCModule.sendEvent("peerConnectionOnRemoveTrack", params);
+        
+        params.putBoolean("mute", true);
+        webRTCModule.sendEvent("mediaStreamTrackOnMuteChanged", params);
+    };
+
+    // This is only added to compile. Plan B is not supported anymore.
+    @Override
+    public void onRemoveStream(MediaStream stream) {
+
+    }
+
+    // This is only added to compile. Plan B is not supported anymore.
+    @Override
+    public void onAddStream(MediaStream stream) {
+
+    }
+
+    /**
+     * Removes a specific local <tt>MediaStream</tt> from the associated
+     * <tt>PeerConnection</tt>.
+     *
+     * @param localStream the local <tt>MediaStream</tt> from the associated
+     *                    <tt>PeerConnection</tt>
+     * @return <tt>true</tt> if removing the specified <tt>mediaStream</tt> from
+     * this instance resulted in a modification of its internal list of local
+     * <tt>MediaStream</tt>s; otherwise, <tt>false</tt>
+     */
+    boolean removeStream(MediaStream localStream) {
```

Comment:
Do we need this?

Reply:
It turns out, no. We are not modifying storing local media streams inside the observer anymore.